src/shared/Core.UI/Commands/CredentialsCommand.cs

```diff
@@ -63,7 +63,9 @@ private async Task<int> ExecuteAsync(CommandOptions options)
 
             if (!viewModel.WindowResult)
             {
-                throw new Exception("User cancelled dialog.");
+                var message = "User cancelled dialog.";
+                Context.Trace2.WriteError(message);
+                throw new Exception(message);
```

Comment:
Would it help (or would this be a hack) if we made `Context.Trace2.WriteError(m)` return `m`, so that you could do:

```
throw new Exception(Context.Trace2.WriteError(message))`
```

That might be a style clash though, so feel free to disregard.

Reply:
I'd like to keep them separate - I think it's a little more readable that way and I'd like to stay consistent with the other `Trace2` methods that don't return anything.